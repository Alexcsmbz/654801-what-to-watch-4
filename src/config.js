export const moviePageTabs = [
  {
    name: `Overview`,
  },
  {
    name: `Details`,
  },
  {
    name: `Reviews`,
  },
];

export const RoutePath = {
  MAIN: `/`,
  LOGIN: `/login`,
  MOVIE: `/films/:id`,
  MYLIST: `/mylist`,
  REVIEW: `/films/:id/review`,
};
