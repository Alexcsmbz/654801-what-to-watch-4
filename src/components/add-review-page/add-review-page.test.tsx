import AddReviewPage from './add-review-page';
import {testMovie, testUser, testComment} from 'config';
import {BrowserRouter as Router} from 'react-router-dom';

jest.mock(`react-router-dom`, () => ({
  ...jest.requireActual(`react-router-dom`),
  useHistory: () => ({
    push: jest.fn()
  })
}));

describe(`App snapshot test`, () => {
  it(`App should render app component`, () => {
    const tree = renderer.create(
        <Router>
          <AddReviewPage
            user={testUser}
            movie={testMovie}
            sendReview={() => { }}
            setReview={() => { }}
            review={testComment}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
