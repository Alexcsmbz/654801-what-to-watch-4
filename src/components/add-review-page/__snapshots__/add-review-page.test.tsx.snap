// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test App should render app component 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
      style={
        Object {
          "backgroundColor": "backgroundColor",
        }
      }
    >
      <img
        alt="name"
        src="backgroundImage"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link undefined"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/1"
              onClick={[Function]}
            >
              name
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <a
            href="/mylist"
            onClick={[Function]}
          >
            <img
              alt="name avatar"
              height="63"
              src="https://4.react.pages.academyavatarUrl"
              width="63"
            />
          </a>
        </div>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="name"
        height="327"
        src="posterImage"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      className="add-review__form"
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            type="radio"
            value={1}
          />
          <label
            className="rating__label"
            htmlFor="star-1"
            onClick={[Function]}
          >
            Rating 
            1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            type="radio"
            value={2}
          />
          <label
            className="rating__label"
            htmlFor="star-2"
            onClick={[Function]}
          >
            Rating 
            2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            type="radio"
            value={3}
          />
          <label
            className="rating__label"
            htmlFor="star-3"
            onClick={[Function]}
          >
            Rating 
            3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            type="radio"
            value={4}
          />
          <label
            className="rating__label"
            htmlFor="star-4"
            onClick={[Function]}
          >
            Rating 
            4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            type="radio"
            value={5}
          />
          <label
            className="rating__label"
            htmlFor="star-5"
            onClick={[Function]}
          >
            Rating 
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="sc-AxjAm kizFCO add-review__btn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
